### ISSUE TITLE
Use pubsub and blocking for a quicker roundtrip

### ISSUE THREAD

**Issue by m-kuhn** [samuelcolvin/arq#404]
arq currently uses a polling mode (default 0.5 seconds)  to fetch jobs on worker side and also to fetch results.
I think it should be possible to use blocking requests (on worker side) and pubsub (possibly with keyspace notifications, but not necessarily) for fetching the result.

This should be quite reliable in most scenarios, however it can occur that a job gets lost (there is no guarantee for pubsub delivery after all), so one could still poll from time to time to be on the safe side.

Are there any opinions on development in this direction?

**Comment by m-kuhn** [comment]
CC @arturzx

**Comment by gg718** [comment]
See also how `rq` recently implemented this: https://github.com/rq/rq/pull/1939

**Comment by JonasKs** [comment]
A draft PR welcome.😊

**Comment by samuelcolvin** [comment]
My plan is to move to redis streams which are by far the best solution for tasks like those performed by arq. Stay tuned for a big update.

**Comment by samuelcolvin** [comment]
See #437.

**Comment by Wh1isper** [comment]
I agree that redis stream is the best option right now, can't wait for the arq update, I developed a redis stream based library, aka [brq](https://github.com/Wh1isper/brq). If you're in a hurry, you can try that too.
